### PR TITLE
feat(context-menu-item): add support for danger kind

### PR DESCRIPTION
**Pull request by janhassel**
Adds support for a danger kind for `ContextMenuItem` which receives a red (`$danger-01`) background on hover and focus, similar to the OverflowMenuItem.

#### Changelog

**New**

- `kind` prop for `ContextMenuItem`
- danger styles for `ContextMenuItem`

**Changed**

- Updated story to use `danger` kind on the "Remove" item.

#### Testing / Reviewing

- On focus and hover, the "Remove" item in the story should receive a `$danger-01` background and `$text-04` color
- If `props.kind="danger"` is passed to an item and it has children, the danger styles are ignored
  - In the demo, if one would pass `props.kind="danger"` to the first item ("Share with"), it's not gonna be red as it has a submenu

image

### CODE REVIEW

**Review by joshblack — APPROVED** [review]

**Review by dakahn — APPROVED** [review]

**Comment by dakahn** [comment]
@carbon-design-system/design tapping yall in to 👍🏽 these changes

tldr adds a danger state and style to the buttons in the overflow menu

**Review by aagonzales — APPROVED** [review]
This additional seems fair, looks good! (I didn't even realize context menu was up in the experimental section either lol 🎉 )

**Comment by janhassel** [comment]
recheck
